### PR TITLE
Issue/16 material design

### PR DESCRIPTION
**Pull request by tonyr59h**
Updating keyboard and preferences to follow Material design guidelines more closely.

Layouts for larger screens limits the size of the keypad, it was hideously huge before.
Ripple effect added to keypad buttons.
Removed the WordPress logo in favor of something more generic.

After the merge we'll need to bump the version and do a new release so the changes can be pulled into the WordPress-Android project.

cc @roundhill

### CODE REVIEW

**Comment by roundhill** [comment]
What if we used this icon from the material icon set in place of the 'Del' key? Maybe it'd be harder to theme, though...

ic_backspace_black_24dp

File: library/res/layout-sw600dp/app_passcode_keyboard.xml [hunk 1]
```diff
@@ -1,78 +1,62 @@
...
-                style="@style/PasscodeInputItemStyle" >
-            </EditText>
-
-            <EditText
-                android:id="@+id/pincode_4"
-                style="@style/PasscodeInputItemStyle" >
-            </EditText>
-        </LinearLayout>
+            android:layout_marginRight="38dp"
+            android:hint="@string/passcodelock_hint"
```
Comment (roundhill) [inline library/res/layout-sw600dp/app_passcode_keyboard.xml:50]: Do you think we even need the hint? It says 'Enter PIN code' already right above it.

Comment (roundhill) [inline library/res/layout-sw600dp/app_passcode_keyboard.xml:50]: Or we could kill the label that's above it and just use 'Enter PIN code' as the hint?

**Comment by tonyr59h** [comment]
> What if we used this icon from the material icon set in place of the 'Del' key?

Much preferred over the existing 'Del' text.

**Comment by daniloercoli** [comment]
Could you please update the instructions to change the logo:

<blockquote>Customization
If you want to customize the pinlock icon, the one available in the unlock screen, add a file named passcode_logo.png in res/drawable.
</blockquote>

Also, the default logo doesn't look so good on hi-res devices
screen shot 2015-05-25 at 11 43 50

**Comment by daniloercoli** [comment]
The "Ripple effects" for buttons on the 1st row doesn't have the correct z-index:

passcode-ripple-issue-01

**Comment by daniloercoli** [comment]
One more things on buttons, The ripple effects is not available in the library. I've to apply the WordPress Calypso style to get it working. Maybe worth to style the buttons in the library too.

**Comment by tonyr59h** [comment]
> The "Ripple effects" for buttons on the 1st row doesn't have the correct z-index

After looking at the rest of the WordPress-Android project I've decided to confine the ripple animation within the view. You can see this effect in the MasterBar and Settings pages (and more, I'm sure). With that change in place the z-order is no longer an issue.

> I've to apply the WordPress Calypso style to get it working.

I think this is because we have not defined the colors used by the animation. I've added the required colors and the compatibility library.

**Comment by daniloercoli** [comment]
Not sure if it's an issue of the library or of the app, but the toolbar is missing of the back/home button in `PasscodePreferencesActivity`
screen shot 2015-05-27 at 11 02 47

**Comment by daniloercoli** [comment]
I've investigated a bit about the missing Toolbar in PasscodePreferencesActivity and seems there isn't a n easy solution for that. The simplest I've found is this one: http://mrengineer13.com/how-to-add-an-actionbar-to-your-settingsactivity-with-appcompat-v21/ or rewriting PasscodePreferencesActivity with ActionBarActivity  + PreferenceFragment. 

/cc @maxme

**Comment by maxme** [comment]
> PasscodePreferencesActivity with ActionBarActivity + PreferenceFragment.

Probably the best solution. That's what we did in wpandroid for the same reason.

**Comment by daniloercoli** [comment]
I just found out that the toolbar appears in the app if I removed the 
`android:theme="@style/CalypsoTheme"` to the PasscodePreferencesActivity. 
Maybe it's not a problem of the library, but something we need to tackle in the app.

screen shot 2015-05-27 at 15 26 46

**Comment by tonyr59h** [comment]
I've styled it in the WordPress-Android project to better match the other preference screens.

> PasscodePreferencesActivity with ActionBarActivity + PreferenceFragment

If we want to refine it more and implement it like this I'm not opposed. But in my opinion the style in WPAndroid is very close now.

**Comment by daniloercoli** [comment]
Styling the Preference ActionBar in wp-android seemed to work fine at first glance, but I found issues with the Ripple effect, the label, and the icon. 
Icon - It's visible in the bar. 
Label - It's clickable in PinLock Screen, while it's not in the app.

passcode-ripple-issue-02

Also, introducing new styles that are just used in one PinLock screen seems a bit overkill to me. If we need to change the Action bar in the app, we then need to update these new styles...(i'm not super expert with styles and UI stuff, maybe this is the only way to achieve the goal).

My impressions here is that we need to switch to 

> PasscodePreferencesActivity with ActionBarActivity + PreferenceFragment

as @maxme suggested.

I'd to hear what @nbradbury has to say about this.

**Comment by nbradbury** [comment]
> I'd [like] to hear what @nbradbury has to say about this.

Trying to match the style will likely be an endless battle. Would it be possible to move "Turn PIN lock on" and "Change PIN" to our main preferences screen? That way the only additional screen would be to change the PIN, which is fullscreen and doesn't need to match.

**Comment by tonyr59h** [comment]
@nbradbury thanks for the input, seems we've been looking at it at the wrong angle. I agree, it's usually a good idea to remove a level of indirection with the UI. I'll try moving those options up a level unless there's any objections.

**Comment by daniloercoli** [comment]
> I'll try moving those options up a level unless there's any objections.

:+1:

**Comment by maxme** [comment]
I added sonatype configuration to that repo in this branch, so we can publish the artifact on maven central.

**Comment by daniloercoli** [comment]
I've updated the file readme.md to reflect the switch from PreferenceActivity to PreferenceFragment.

@maxme Can you take a look it? If everything is OK for you,  we're ready to merge this PR, and publish the new lib version 0.0.7 to Maven.

**Comment by tonyr59h** [comment]
Thanks for updating the docs guys! I'll go ahead with the merge and push a new release to Maven Central so we can do a 4.2 code freeze.

**Pull request merged**